hazelcast/src/test/java/com/hazelcast/spi/OperationSerializationTest.java

```diff
@@ -157,23 +163,73 @@ public void assertSerializationCloneEquals(Operation expected) {
         assertEquals("wait timeout does not match", expected.getWaitTimeout(), actual.getWaitTimeout());
     }
 
+    @Test
+    public void test_serviceName_whenOverridesGetServiceName_thenNotSerialized(){
+        OperationWithServiceNameOverride op = new OperationWithServiceNameOverride();
+        assertNull(op.getRawServiceName());
+        assertFalse("service name should not be set", op.isFlagSet(Operation.BITMASK_SERVICE_NAME_SET));
+
+        Operation copy = copy(op);
+        assertSame(DUMMY_SERVICE_NAME, copy.getServiceName());
+        assertNull(copy.getRawServiceName());
+        assertFalse("service name should not be set", copy.isFlagSet(Operation.BITMASK_SERVICE_NAME_SET));
+    }
+
+    @Test
+    public void test_serviceName_whenNotOverridesServiceName_thenSerialized(){
+        DummyOperation op = new DummyOperation();
+        op.setServiceName(DUMMY_SERVICE_NAME);
+        assertSame(DUMMY_SERVICE_NAME, op.getRawServiceName());
+        assertSame(DUMMY_SERVICE_NAME, op.getServiceName());
+        assertTrue("service name should not be set", op.isFlagSet(Operation.BITMASK_SERVICE_NAME_SET));
```

Comment:
To be or not to be! You have never experienced Shakespeare until you have read him in the original Klingon.

(I think you want to have to other one in the assert message)